src/meta/processors/job/JobManager.cpp

```diff
@@ -42,7 +43,10 @@ JobManager* JobManager::getInstance() {
   return &inst;
 }
 
-bool JobManager::init(nebula::kvstore::KVStore* store, AdminClient* adminClient) {
+bool JobManager::init(nebula::kvstore::KVStore* store,
+                      AdminClient* adminClient,
+                      std::shared_ptr<JobExecutorFactory> factory) {
```

Comment:
how about `JobExecutorFactory*` keep consistent with other parameter, besides raw pointer has better performance.

Reply:
> how about `JobExecutorFactory*` keep consistent with other parameter, besides raw pointer has better performance.

Better to make JobManager to control its life. it should not affect performance.

Reply:
Then why not use an  unique_ptr in JobManager, which could make it explicit which lifecycle it belongs.

Reply:
> Then why not use an unique_ptr in JobManager, which could make it explicit which lifecycle it belongs.

I use it for test, test drives me use a shared pointer which I can better mock it's behavior, Maybe unique ptr is better.

Reply:
> Then why not use an unique_ptr in JobManager, which could make it explicit which lifecycle it belongs.

I use it for test, test drives me use a shared pointer which I can better mock it's behavior, Maybe unique ptr is better.